nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/AssumeRoleCredentialsStrategy.java

```diff
@@ -113,16 +134,34 @@ public AWSCredentialsProvider getDerivedCredentialsProvider(Map<PropertyDescript
         rawMaxSessionTime = (rawMaxSessionTime != null) ? rawMaxSessionTime : MAX_SESSION_TIME.getDefaultValue();
         final Integer maxSessionTime = Integer.parseInt(rawMaxSessionTime.trim());
         final String assumeRoleExternalId = properties.get(ASSUME_ROLE_EXTERNAL_ID);
+        STSAssumeRoleSessionCredentialsProvider.Builder builder;
+
+        // If proxy variables are set, then create Client Configuration with those values
+        if (proxyVariablesValidForAssumeRole(properties)) {
+            final String assumeRoleProxyHost = properties.get(ASSUME_ROLE_PROXY_HOST);
+            final Integer assumeRoleProxyPort = Integer.parseInt(properties.get(ASSUME_ROLE_PROXY_PORT));
+            ClientConfiguration config = new ClientConfiguration();
+            config.withProxyHost(assumeRoleProxyHost);
+            config.withProxyPort(assumeRoleProxyPort);
+            AWSSecurityTokenService securityTokenService = new AWSSecurityTokenServiceClient(config);
```

Comment:
That could be true, I'm not sure. I'm running another build now to test if everything will still work if I add `primaryCredentialsProvider` into the `STSAssumeRoleSessionCredentialsProvider` as well.